src/main/christophermanahan/github/io/carnitas/EchoServer.java

```diff
@@ -0,0 +1,31 @@
+package christophermanahan.github.io.carnitas;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class EchoServer implements Server {
+
+  final ServerSocket serverSocket;
+
+  EchoServer(ServerSocket serverSocket) {
+    this.serverSocket = serverSocket;
+  }
+
+  public void run() {
+    try {
+      Socket socket = serverSocket.accept();
+
+      Connection socketConnection = new SocketConnection(socket);
+      Input socketInput = new SocketInput(socket);
+      Output socketOutput = new SocketOutput(socket);
```

Comment:
I believe it depends on the role of a connection. If I define the responsibility of a connection as managing the operations associated with itself, ie. send, receive, open, and close than the input and output do belong in the abstraction. It feels like there may be some subjectivity involved in interpreting a single responsibility?

Reply:
>It feels like there may be some subjectivity involved in interpreting a single responsibility?
Certainly. But it isn't just subjectivity here. We have also completely changed the abstraction.

Reply:
I believe that it may be more worth my time to focus on the other code review comments so that I can complete the work for this iteration. I can start out next week with refactoring the connection to have the ability to send, receive, open, and close. For this weeks iteration, I don’t think refactoring so late in the iteration is a good idea if I want a demo ready tomorrow.